testplan/exporters/testing/http/__init__.py

```diff
@@ -14,6 +16,12 @@
 from ..base import Exporter
 
 
+class CustomJsonEncoder(json.JSONEncoder):
```

Comment:
Why is this needed exactly?

Reply:
user can define attributes in test suites, but we don't have a check that all their attributes can be serialized by JSON (this rarely happens, but user can define weird things like some objects of non-basic type). To avoid exception in dumping JSON so we can add this customized encoder. Xml pdf exported is not affected because they choose the pre-defined attributes to dump. I think Json report is the most basic if we want to debug or trace, so it should contains all the information. Json exporter also need it so I move this Encoder to exporters/testing/json/__init__.py and HTTPExporter imports it.

Reply:
ok, I think that if a user defines some weird attribute on a testsuite that is quite unusual and we shouldn't worry about making it work. But if it's only a simple change it's fine to leave in.

Reply:
It's only a few lines change, more safe, some team like proxima like to dynamically throw things into modules or classes. Let me keep it.